torch/_functorch/aot_autograd.py

```diff
@@ -2634,6 +2736,10 @@ def create_aot_dispatcher_function(
     if aot_config.decompositions is None:
         aot_config.decompositions = {}
 
+    if config.functionalize_rng_ops:
+        # Update the decompositions with functionalized random decompositions
+        aot_config.decompositions.update(rng_decompositions)
```

Comment:
I don't think you're allowed to mutate the input decomp table.  Instead, merge them in on the lines below.